### DISCUSSION TITLE
Conditionally turn a Hue smart plug device on / off via service

### DISCUSSION THREAD

**Discussion by lougreenwood** [zachowj/node-red-contrib-home-assistant-websocket#835] [Q&A] [answered]
This seem like it should be trivial, but I can't figure out how to do this!

I have a service node, it is configured in the node config UI to control a hue light (which is actually a smart plug, Hue just exposes smart plugs as lights).

![Screenshot 300](https://user-images.githubusercontent.com/12051736/211356923-7d22af28-f583-4334-bbdc-93012ad2ba13.png)

I want to conditionally turn the "light" on / off depending on the value sent to the service.

It seem that the `toggle` service isn't suitable, since it's description mentions "Toggles one or more lights, from on to off, or, off to on, **based on their current state**." (i.e it implies it can't be conditionally controlled, it toggles to the opposite of it's previous state).

Also neither `turn_on` / `turn_off` seem suitable, since they imply that the are mutually exclusive controls and I would need 2 service nodes to handle it, in addition to routing to choose the correct service node.

Then... once the above is figured out, I cannot figure out how to send a payload to the node to get it to send the trigger. The docs talk about the `data` attribute being a json object, but when ever I try to include an object with params, I get the following error: `"Call-service error. extra keys not allowed @ data['title']"`

I've tried just sending the data as a bool or string (`on` / `off`, like what I receive as the state value on a `state_changed` event when I externally control this device), but this doesn't seem to work...

I hate to ask for hand-holding help, but could someone point me to how I send a conditional instruction to turn a light on / off. So far everything seems overly complex for such a simple use case. The docs for the service node are cryptic and don't include a trivial example of how to send a payload to control a device / entity... The examples in the node are missing any attribute to control the light `on` / `off` state.

![Screenshot 302](https://user-images.githubusercontent.com/12051736/211357534-08920187-dd2b-47c8-b78d-4d63c24b6457.png)

(I also tried the `device` node, but it seems I hit the issue above of which `trigger` to use: `toggle`, `turn on` or `turn off` - none seem suitable.... and in any case, what should the msg object look like to trigger a `device` node, there are no docs for this).

Sorry if that came across as ranty, I love the addon - I'm currently trying to migrate my pretty complex light home automation system with over 50 lights, sensors and complex conditional logic, but I'm feeling frustrated that such a simple thing is tripping me up! 😢

**Comment by zachowj** [comment, marked as answer]
Does this example help?

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/call-service.html#using-mustache-templates>

**Comment by lougreenwood** [comment]
Sorry for the delay in replying, but yes, thanks @zachowj!
